packages/editor/src/components/collab-sidebar/hooks.js

```diff
@@ -21,19 +21,25 @@ export function useBlockComments( postId ) {
 		{ enabled: !! postId && typeof postId === 'number' }
 	);
 
-	const blocksWithComments = useSelect( ( select ) => {
-		const { getBlockAttributes, getClientIdsWithDescendants } =
-			select( blockEditorStore );
+	const { getBlockAttributes } = useSelect( blockEditorStore );
+	const { clientIds } = useSelect( ( select ) => {
+		const { getClientIdsWithDescendants } = select( blockEditorStore );
 
-		return getClientIdsWithDescendants().reduce( ( results, clientId ) => {
+		return {
+			clientIds: getClientIdsWithDescendants(),
+		};
+	}, [] );
+
+	const blocksWithComments = useMemo( () => {
+		return clientIds.reduce( ( results, clientId ) => {
```

Comment:
@adamsilverstein, here you can return more complex object shapes if you want, in case that helps with lookup when sorting the comments.